packages/api/src/controllers/usage.js

```diff
@@ -0,0 +1,89 @@
+import Router from "express/lib/router";
+import { db } from "../store";
+
+const app = Router();
+
+app.get("/", async (req, res) => {
+  if (!req.headers.authorization) {
+    res.status(403);
+    return res.json({ errors: ["unauthorized"] });
+  }
+
+  let token = req.headers.authorization.split(" ")[1]; // Bearer <token>
+
+  if (process.env.LP_API_ADMIN_TOKEN != token) {
+    res.status(403);
+    return res.json({ errors: ["unauthorized"] });
+  }
+
+  let { fromTime, toTime } = req.query;
+
+  // if time range isn't specified return all usage
+  if (!fromTime && !toTime) {
+    fromTime = +new Date(2020, 0); // start at beginning
+    toTime = +new Date();
+  }
+
+  const cachedUsageHistory = await db.stream.cachedUsageHistory(
+    fromTime,
+    toTime,
+    {
+      useReplica: true,
+    }
+  );
+
+  res.status(200);
+  res.json(cachedUsageHistory);
+});
+
+app.post("/update", async (req, res) => {
+  if (!req.headers.authorization) {
+    res.status(403);
+    return res.json({ errors: ["unauthorized"] });
+  }
+
+  let token = req.headers.authorization.split(" ")[1]; // Bearer <token>
+
+  if (process.env.LP_API_ADMIN_TOKEN != token) {
+    res.status(403);
+    return res.json({ errors: ["unauthorized"] });
+  }
+
+  let fromTime = +new Date(2020, 0); // start at beginning
+  let toTime = +new Date();
+
+  let cachedUsageHistory = await db.stream.cachedUsageHistory(
```

Comment:
Instead of having to run a query that fetches all rows since the beginning for each update could it make sense to have another query + function that just returns the last update time?